src/Entity/PrintArticleType.php

```diff
@@ -174,17 +185,64 @@ public function buildThumbnail() {
   /**
    * {@inheritdoc}
    */
+  public function save() {
+    $violations = $this->validate();
+    if ($violations->count()) {
+
+      /** @var \Symfony\Component\Validator\ConstraintViolationInterface $violation */
+      foreach ($violations as $violation) {
+        throw new \Exception($violation->getMessage());
+      }
+    }
+    return parent::save();
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @see \Drupal\Core\Entity\ContentEntityBase::preSave()
+   */
   public function preSave(EntityStorageInterface $storage) {
+    // The entity should not be saved, unless it was validated succesfully.
+    if (!$this->validated) {
+      throw new \LogicException('Entity validation was skipped.');
+    }
+    // We reset validated to FALSE, so any further changes would end up in
+    // checking validation again.
+    else {
+      $this->validated = FALSE;
+    }
+
     parent::preSave($storage);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate() {
+    $this->validated = TRUE;
 
     $idms = new IDMS($this->idms);
 
-    /** @var \Symfony\Component\Validator\ConstraintViolationListInterface $errors */
-    $errors = $idms->validate();
+    // Validate IDMS.
+    $idmsViolations = $idms->validate();
+
+    // Validate $this.
+    $printArticleTypeViolations = Validation::createValidator()->validate($this, [
+      new IdmsUniqueTags(),
```

Comment:
XMLTag/Caption is ok. We have to check everything in the Story tag. XMLTag/Caption appears more often, but thats in an XMLStory element.

You can check it with Zeitung2 and XMLTag/Textabschnitt. I am checkin' that in testMultipleTagsException as well